### ISSUE TITLE
Warn against using data types that have no JSON equivalent in attributes?

### ISSUE THREAD

**Issue by mconigliaro** [acrmp/foodcritic#286]
It would be nice to have a rule to warn against using data types that have no JSON equivalent in attributes. Unless there's some way to encode object information that I've missed? For example:

``` ruby
# recipes/example.rb
node.default['foo'] = :bar
case node['foo']
when :bar 
  ...
when :baz
  ...
end
```

This works fine if you're only dealing with Ruby inside cookbooks, but the problem comes up when you try to override attributes in Chef environments/roles or inside `.kitchen.yml`. For example:

``` ruby
# roles/example.rb
name 'example'
default_attributes('foo' => :baz)
```

This role will be converted to JSON before being uploaded to the server, but since JSON has no concept of symbols, there doesn't seem to be any way to override `node['foo']` with `:baz`. In this case, the symbol `:baz` will get converted to **string** `"baz"`, which may make your code fail in unexpected ways.

**Comment by lamont-granquist** [comment]
Yeah good point, if you round trip it to JSON and back and don't get the same thing, then it should warn on that.  Catching `node#[]=` being passed a bad value is probably easier to parse.  Catching arbitrary ways of comparing node data against invalid data types is probably going to be pretty hard.
